src/App.tsx

```diff
@@ -8,32 +10,45 @@ function getRandomName(): string {
 }
 
 export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
+  const [clockName, setClockName] = useState('Clock-0');
+  const [hasClock, setHasClock] = useState(true);
+
+  const onRightClick = (event: MouseEvent) => {
+    event.preventDefault();
+    setHasClock(false);
+  };
+
+  const onClick = () => setHasClock(true);
+
+  useEffect(() => {
+    if (hasClock && clockName !== 'Clock-0') {
+      console.debug(`Renamed to ${clockName}`);
+    }
+  }, [clockName]);
+
+  useEffect(() => {
+    document.addEventListener('click', onClick);
+    document.addEventListener('contextmenu', onRightClick);
+
+    const timerId = window.setInterval(
+      () => setClockName(getRandomName()),
+      3300,
+    );
+
+    return () => {
+      document.removeEventListener('click', onClick);
+      document.removeEventListener('contextmenu', onRightClick);
+      window.clearInterval(timerId);
+    };
```

Comment:
this is an excellent job!, you remembered to removing the eventListener , Keep doing this!